### PR TITLE
Move fixtures in specific package

### PR DESCRIPTION
**Pull request by edvgui**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -27,13 +28,22 @@ apispec = "^5.1.1"
 requests = "^2.23.0"
 pydantic = "^1.8.2"
 
+docker = { version = "^5.0.3", optional = true }
+MarkupSafe = "2.0.1"
+
+[tool.poetry.extras]
+pytest = ["docker"]
```
Comment (sanderr) [inline pyproject.toml:35]:
```
>>> import pytest_nfv_test_api
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/sander/.virtualenvs/tmp-ec786a97787c754/lib/python3.9/site-packages/pytest_nfv_test_api/__init__.py", line 25, in <module>
    import pytest
ModuleNotFoundError: No module named 'pytest'
>>> 
```
```suggestion
pytest = ["docker", "pytest"]
```
You'll also need to add it to the dependencies list of course.

File: tests/contest.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+"""
```
Comment (sanderr) [inline tests/contest.py:1]: I think you can safely delete this file.

**Review by sanderr — CHANGES_REQUESTED** [review]

File: .bumpversion.cfg [hunk 3]
```diff
@@ -0,0 +1,20 @@
+[bumpversion]
+current_version = 2.1.0
+tag = False
+commit = False
+
+[bumpversion:file:pyproject.toml]
+search = 
+	name = "inmanta-nfv-test-api"
+	version = "{current_version}"
+replace =
```
Comment (sanderr) [inline .bumpversion.cfg:10]:
```suggestion
replace =
```

File: .bumpversion.cfg [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+[bumpversion:file:pyproject.toml]
+search = 
+	name = "inmanta-nfv-test-api"
+	version = "{current_version}"
+replace = 
+	name = "inmanta-nfv-test-api"
+	version = "{new_version}"
+
+[bumpversion:file:src/nfv_test_api/__init__.py]
```
Comment (sanderr) [inline .bumpversion.cfg:14]: Not sure these files require custom search and replace but it can do no harm either.

File: .ci/ci-pyp.yml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+pipeline_template: ci/inmanta-tools/Jenkinsfile-pyp
+repo_name: inmanta-nfv-test-api
+public: true
+build_system: pep517
```
Comment (sanderr) [inline .ci/ci-pyp.yml:4]: `pep517` has been deprecated. In the pipeline it is now a synonim for "generic", which uses `python -m build`.
```suggestion
build_system: generic
```

File: .ci/Jenkinsfile-integration-test [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+                        passwordVariable: 'DEVPI_PASS',
+                        usernameVariable: 'DEVPI_USER'
+                    )
+                    ]) {
+                    sh '''
+                        ${WORKSPACE}/env/bin/pip3 install -U devpi-client
+                        ${WORKSPACE}/env/bin/devpi use https://artifacts.internal.inmanta.com/inmanta/dev
+                        ${WORKSPACE}/env/bin/devpi login ${DEVPI_USER} --password=${DEVPI_PASS}
+                        rm -f dist/*
+                        ${WORKSPACE}/env/bin/python3 -m poetry build
```
Comment (sanderr) [inline .ci/Jenkinsfile-integration-test:94]: Does this produce artifacts with the same naming scheme as the nightly ones?

File: .bumpversion.cfg (hunk 4 above)
Comment (edvgui) [reply .bumpversion.cfg:14]: How else would you do it?

Comment (sanderr) [reply .bumpversion.cfg:14]: The default just searches for the version and replaces it with the version so you could just leave the sections empty.

Comment (edvgui) [reply .bumpversion.cfg:14]: Oh, nice

File: .ci/Jenkinsfile-integration-test (hunk 6 above)
Comment (edvgui) [reply .ci/Jenkinsfile-integration-test:94]: What do you mean with nightly ones?

File: src/pytest_nfv_test_api/__init__.py [hunk 7]
```diff
@@ -45,25 +47,25 @@ def free_image_tag(docker_client: docker.DockerClient) -> Generator[str, None, N
...
 @pytest.fixture(scope="session")
-def nfv_test_api_image(docker_client: docker.DockerClient, free_image_tag: str) -> Generator[images.Image, None, None]:
+def nfv_test_api_image(docker_client: docker.DockerClient, free_image_tag: str) -> images.Image:
     """
     This fixture builds a container image containing the nfv-test-api server.
     """
     try:
         docker_client.images.build(
-            path=os.getcwd(),
+            path=os.path.realpath(os.path.join(__file__, "../../..")),
```
[the remark this answers is not in the document]
Comment (edvgui) [reply src/pytest_nfv_test_api/__init__.py]: Right, I don't think it ever will be used on non-unix hosts, but I changed it anyway

File: .ci/Jenkinsfile-integration-test (hunk 6 above)
Comment (sanderr) [reply .ci/Jenkinsfile-integration-test:94]: The ones built via [.ci/ci-pyp.yml](https://github.com/inmanta/nfv-test-api/pull/44/files#diff-fde24906043bd56abd1098810f4563ea781795a292d6ab746bd42847ac31fcfe).

Comment (sanderr) [reply .ci/Jenkinsfile-integration-test:94]: I guess it does. I don't think the pipeline sets custom tags for the generic build backend.

**Pull request merged**